File: app/src/at-trinhnguyen/java/asiantech/internship/winter/javacore/oop/Rectangle.java
package com.congtrinh.Bai32;

public class Rectangle extends Shape {
    private double length ;
    private double width;

    @Override
    public void print() {
        if (length != width) {
            System.out.print("Rectangle ");
        }else super.print();
    }


    public Rectangle(double length, double width) {
        this.length  = length;
        this.width = width;
    }

    @Override
    public double area() {
        return length  * width;
    }

    @Override
    public double circumference() {
        return (length  + width) * 2;
    }
}

File: app/src/at-trinhnguyen/java/asiantech/internship/winter/javacore/oop/Rectangle.kt
package asiantech.internship.winter.javacore.oop

class Rectangle(private val length: Double, private val width: Double) : Shape() {

    public override fun print() {
        if (length != width) {
            print("Rectangle ")
        } else
            super.print()
    }

    public override fun area(): Double {
        return length * width
    }

    public override fun circumference(): Double {
        return (length + width) * 2
    }
}
